src/Service.GraphQLBuilder/Mutations/MutationBuilder.cs

```diff
@@ -53,7 +58,17 @@ public static class MutationBuilder
 
                         if (isSPDefinedAsMutation)
                         {
-                            AddMutationsForStoredProcedure(dbEntityName, entityPermissionsMap, name, entities, mutationFields);
+                            if (dbObjects is not null && dbObjects.TryGetValue(entityName, out DatabaseObject? dbObject) && dbObject is not null)
+                            {
+                                AddMutationsForStoredProcedure(dbEntityName, entityPermissionsMap, name, entities, mutationFields, dbObject);
+                            }
+                            else
+                            {
+                                throw new DataApiBuilderException(
+                                    message: "GraphQL Schema Creation for Stored Procedures requires DatabaseObject defined. Most likely error in startup.",
```

Comment:
Is there a scenario where we might fail to determine the dbobject definition for the stored procedure even when the stored procedure did exist in the underlying database? If not, how about we use `ServiceUnavailable` statuscode with `ErrorInInitialization` substatuscode?

Reply:
updated , thanks for the suggestion.